akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -346,9 +349,12 @@ final class Split[T](decision: Split.SplitDecision, p: T ⇒ Boolean) extends Gr
             closeThis(handler, elem)
             handOver(handler)
           } else {
-            // Drain into the void
-            if (substreamCancelled) pull(in)
-            else substreamSource.push(elem)
+            if (substreamCancelled) {
+              if (eagerClose) completeStage()
```

Comment:
OK. Should it also be renamed on `outgoingConnection`? I think `eagerClose` suits it better, since it will translate into connection being closed.